dp/cloud/python/magma/configuration_controller/crl_validator/certificate.py

```diff
@@ -0,0 +1,103 @@
+"""
+Copyright 2022 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import socket
+import ssl
+from typing import List
+
+import requests
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from requests.exceptions import SSLError
+
+_x509_backend = default_backend()
+
+
+def get_certificate(
+        hostname: str,
+        port: int = 443,
+) -> x509.Certificate:
+    """ Retrieve certificate of given host.
+
+    Args:
+        hostname: host from which certificate will be retrieved
+        port: SSL port of the host
+
+    Returns:
+        x509.Certificate
+    """
+    context = ssl.SSLContext(protocol=ssl.PROTOCOL_SSLv23)
+    context.minimum_version = ssl.TLSVersion.TLSv1_2
+
+    sock = socket.create_connection((hostname, port))
```

Comment:
What if there is an exception somewhere in the middle of this function?
The socket won't be closed, so I think we use `with`